src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilation.cs

```diff
@@ -29,7 +29,8 @@ public CSharpCompilation(Compilation compilation)
 
         public ImmutableArray<Diagnostic> GetDiagnostics()
         {
-            return _compilation.WithAnalyzers(GetAnalyzers()).GetAllDiagnosticsAsync().Result;
+            var diagnostics = _compilation.WithAnalyzers(GetAnalyzers()).GetAllDiagnosticsAsync().Result;
```

Comment:
Its confusing that you have to do to calls here - why is that? GetAllDiagnostics should, well, get them ALL right?